### DISCUSSION TITLE
Wrong Colors...

### DISCUSSION THREAD

**Discussion by neTix84** [jasoncoon/esp8266-fastled-webserver#245] [Q&A] [answered]
Hello everybody, 

i habe the problem, when i set the solid color to red, the led strip shows green.
Via google I found that it could be that some strips not RGB, they are BGR (some manufacture has different wiring), but I don't know where to change this? Could someone help me?

One other problem, this first led is flickering, also when i turn it of (in the web interface), any idea what could be the reason?

Thanks for your help.
Cheers, 
Danny

**Comment by devoh747** [comment]
It appears that you need to know what Fib product you are compiling for.. look at around line 56-> 82 in https://github.com/jasoncoon/esp8266-fastled-webserver/blob/main/esp8266-fastled-webserver/config.h .  I have a Fib256 so I would then look here  https://github.com/jasoncoon/esp8266-fastled-webserver/blob/main/esp8266-fastled-webserver/include/configs/product/fibonacci256.h  which on line 27 has the color order defined for my Fib256.  Then you would need to recompile it.

Hope this helps.

  -Dave

**Comment by jasoncoon** [comment, marked as answer]
Assuming you're using the default configuration and/or Arduino, you probably just want to edit the default config here: https://github.com/jasoncoon/esp8266-fastled-webserver/blob/main/esp8266-fastled-webserver/include/configs/product/default.h#L27

**Reply by neTix84** [reply]
Thank you both.
@jasoncoon  your assumption is correct.
Thank you. Directly above the other parameters, which I have already adjusted (it was just too late yesterday).

**Comment by neTix84** [comment]
Maybe someone has an idea why the first LED always flickers? The LED flashes again and again in other colors (red & blue), you can see this when you set a solid color or turn off the LED's via controller (then it is most noticeable because all others are off).

**Reply by henrygab** [reply]
There are common causes of the first LED flickering.

Have you tried bypassing that first LED?  the LED could have been damaged.

Are you using a level shifter (SN74HCT245N is commonly used), or directly driving the 5V LED data line using 3V3 output of an ESP8266/ESP32?

Is there a common ground for LEDs and ESP?  Ground must be shared.

Other causes can include voltage drop (insufficient wire gauge / too long wires), damage to the first LED (such as ESD), etc.

**Reply by jasoncoon** [reply]
I recently had the same problem when using the Arduino IDE, newer versions of the ESP8266 board, and FastLED 3.4.  You might try again with ESP8266 board version 2.7 and/or FastLED 3.5.

**Reply by neTix84** [reply]
Thanks for your feedback. 

@jasoncoon Yes, I used the Arduino IDE, but in the meantime, I use ESPHome for my devices (also with FastLED, don't know the version), but no problems yet.

@henrygab I also used a level shifter and an external power supply.
